components/automate-ui/src/app/page-components/simple-line-graph/simple-line-graph.component.ts

```diff
@@ -0,0 +1,356 @@
+      .tickSizeOuter(0)
+      .scale(this.xScale);
+    // Render the X grid lines
+    const grid = this.svgSelection.selectAll('.grid').data([this.data]);
+    grid.exit().remove();
+    grid.enter().append('g').attr('class', 'grid')
+      .attr('transform', `translate(0, ${this.height - this.margin.bottom})`)
+      .merge(grid).transition().duration(this.t)
+      .call(xGrid);
+
+    // create the Y axis
+    const yAxis = d3.axisRight(this.yScale).tickFormat(d => d + '%').ticks(1);
+    // render the Y axis
+    const y = this.svgSelection.selectAll('.y-axis').data([this.data]);
+    y.exit().remove();
+    y.enter().append('g').attr('class', 'y-axis')
+      .attr('transform', `translate(${this.margin.left}, 0)`)
+      .merge(y).transition().duration(this.t)
+      .call(yAxis);
+
+    // create the X axis
+    const xAxis = d3.axisBottom().ticks(this.data.length)
+      .tickSizeInner(10).tickSizeOuter(0).tickFormat('')
+      .scale(this.xScale);
+    // render the X axis
+    const x = this.svgSelection.selectAll('.x-axis').data([this.data]);
+    x.exit().remove();
+    x.enter().append('g').attr('class', 'x-axis')
+      .attr('transform', `translate(0, ${this.height - this.margin.bottom})`)
+      .merge(x).transition().duration(this.t)
+      .call(xAxis);
+
+    // remove zero from bottom of chart on X axis
+    this.svgSelection.selectAll('.tick')
+      .filter(tick => tick === 0)
+      .remove();
+  }
+
+  private handleHover(d3Event): void {
+    const num = this.getHoveredElement(d3Event);
+    d3.selectAll(`.elem-${num}`).classed('active', true);
+  }
+
+  private handleClick(d3Event): void {
+    const num = this.getHoveredElement(d3Event);
+    const isAlreadyLocked = d3.selectAll(`.elem-${num}`).classed('lock');
+    if (isAlreadyLocked) {
+      d3.selectAll(`.elem-${num}`).classed('lock', false);
+      this.locked = null;
+    } else {
+      d3.selectAll('.lock').classed('lock', false);
+      d3.selectAll(`.elem-${num}`).classed('lock', true);
+      this.locked = num;
+    }
+  }
+
+  private getHoveredElement(d3Event): number {
+    const classes = d3.select(d3Event.target).attr('class');
+    const match = classes.match(/elem-([0-9]{1,2})/g)[0];
+    const num = match.split('-')[1];
+    return num;
+  }
+
+  private relock(): void {
+    if (this.locked) {
+      d3.selectAll(`.elem-${this.locked}`).classed('lock', true);
+    }
+  }
+
+  private AllDeactivate(): void {
+    d3.selectAll('.active').classed('active', false);
+  }
+
+  private allUnlock(): void {
+    d3.selectAll('.lock').classed('lock', false);
+  }
+
+  private AllUnlockDeactivate(): void {
+    this.allUnlock();
+    this.AllDeactivate();
+  }
+
+  onResize(): void {
+    this.t = 0; // turn off transitions for resize events;
+    this.renderChart();
+  }
+
+  private resizeChart(): void {
+    this.width = this.chart.nativeElement.getBoundingClientRect().width;
+  }
+
+  ngOnInit() {
```

Comment:
By convention, should put `ngOnInit` and `ngOnChanges` as close to the top of the class as possible.